kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/db/H2DbServiceImpl.java

```diff
@@ -279,6 +303,12 @@ private void updateInternal(final Map<String, Object> properties) {
                 restartDefragTask(newConfiguration);
             }
 
+            if (this.executorService != null) {
+                this.executorService.shutdown();
+                awaitExecutorServiceTermination();
```

Comment:
The current thread is holding `this.rwLock.writeLock`, there might be tasks in the executor that are waiting to acquire a `this.rwLock.readLock`. If this happens, `awaitExecutorServiceTermination()` will not return before the 1 minute timeout and these tasks will run after `awaitExecutorServiceTermination()` returns.